main.js

```diff
@@ -0,0 +1,142 @@
+// The app collection
+const AppCollection = Backbone.Collection.extend({
+  // grabs the url thats passed down from the input
+  url() {
+    return this.models[0].attributes.url;
+  },
+  // on the first instance on the add event do the api fetch
+  initialize() {
+    this.on('add', function (model) {
+      model.fetch();
+    });
+  },
+});
+
+// the App Model that references the collection
+const AppModel = Backbone.Model.extend({
+  defaults() {
+    return {
+      videos: new AppCollection(),
+    };
+  },
+});
+
+// the App View that grabs the input and renders the clickable images
+const AppView = Backbone.View.extend({
+  el: 'body',
+
+  // handlebars template
+  template: Handlebars.compile($('#vid-list').html()),
+
+  // event for the click on the input button invokes the add to model
+  events: {
+    'click .search-btn': 'videoSearchAdd',
+  },
+
+  // on instance grab the input id & listene to changes on the videos model and call renderThumbnails
+  initialize() {
+    this.$searchInput = this.$('#search-input');
+    // this.$vidView = $('.vid-container');?
+    this.listenTo(this.model.get('videos'), 'change', this.renderThumbnails);
+  },
+
+  // this function grabs the value of input adds it to the model so it could be grabbed by the url search
+  videoSearchAdd() {
+    const inputValue = this.$searchInput.val();
+
+    const urlInput = `https://www.googleapis.com/youtube/v3/search?part=snippet&maxResults=5&q=${inputValue}&type=video&videoEmbeddable=true&key=AIzaSyCn7T_poRhx_PIw7R7PrfqQXYuRJH8RQjw`;
```

Comment:
The view should not be concerned with the `url` - the `url` has nothing to do with the user, but rather is how the model should be fetching videos.

The collection should be doing the fetch. In fact, each video model (or whatever we're calling the models that live inside of the videos collection) is getting all 5 videos back from the server. So every time you do a video search, you're hitting the YouTube API 5 times and getting 25 videos (but 5 are duplicates).